src/test/deploy/functions/cel.spec.ts

```diff
@@ -0,0 +1,530 @@
+  });
+
+  describe("Equality expressions", () => {
+    it("raises when the LHS param does not exist", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == 22 }}", {});
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the type of the LHS and RHS do not match", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == 'asdf' }}", {
+          FOO: numberV(22),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("it determines whether the LHS resolves to the same thing as the RHS", () => {
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == 22 }}", {
+          FOO: numberV(22),
+        })
+      ).to.be.true;
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == 11 }}", {
+          FOO: numberV(22),
+        })
+      ).to.be.false;
+      expect(
+        resolveExpression("boolean", '{{ params.FOO == "bar baz" }}', {
+          FOO: new ParamValue("bar baz", false, { string: true }),
+        })
+      ).to.be.true;
+      expect(
+        resolveExpression("boolean", '{{ params.FOO == "baz" }}', {
+          FOO: stringV("bar"),
+        })
+      ).to.be.false;
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == true }}", {
+          FOO: boolV(true),
+        })
+      ).to.be.true;
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == false }}", {
+          FOO: boolV(true),
+        })
+      ).to.be.false;
+    });
+  });
+
+  describe("Dual equality expressions", () => {
+    it("raises when the LHS param does not exist", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == params.BAR }}", {
+          BAR: numberV(22),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the RHS param does not exist", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == params.BAR }}", {
+          FOO: numberV(22),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when a literal is provided as the LHS", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ 22 == 11 }}", {});
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the type of the LHS and RHS do not match", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == params.BAR }}", {
+          FOO: numberV(22),
+          BAR: boolV(true),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("it determines whether the LHS resolves to the same thing as the RHS", () => {
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == params.BAR }}", {
+          FOO: numberV(22),
+          BAR: numberV(22),
+        })
+      ).to.be.true;
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == params.BAR }}", {
+          FOO: numberV(22),
+          BAR: numberV(11),
+        })
+      ).to.be.false;
```

Comment:
also feels a bit sneaky since it doesn't match the test desription.

Reply:
Bad copy-paste job instead of intentionally sneaky, but yes, I'm fixing it